tests/build.go

```diff
@@ -616,6 +618,67 @@ func buildMetadata(t *testing.T, sb integration.Sandbox, metadataMode string) {
 	require.Equal(t, provenancetypes.BuildKitBuildType, prv.BuildType)
 }
 
+func testBuildMetadataStatus(t *testing.T, sb integration.Sandbox) {
+	t.Run("max", func(t *testing.T) {
+		buildMetadataStatus(t, sb, "max")
+	})
+	t.Run("warnings", func(t *testing.T) {
+		buildMetadataStatus(t, sb, "warnings")
+	})
+	t.Run("disabled", func(t *testing.T) {
+		buildMetadataStatus(t, sb, "disabled")
+	})
+}
+
+func buildMetadataStatus(t *testing.T, sb integration.Sandbox, metadataMode string) {
+	if metadataMode != "disabled" {
+		skipNoCompatBuildKit(t, sb, ">= 0.14.0-0", "lint")
```

Comment:
Seems to fail if I don't add this skip. Not sure why marshalling fails in the test :thinking:: https://github.com/docker/buildx/actions/runs/9499025938/job/26179025275#step:7:368

```
=== FAIL: tests TestIntegration/TestBuildMetadataStatus/worker=docker-container/warnings (0.32s)
    build.go:664: 
        	Error Trace:	/src/tests/build.go:664
        	            				/src/tests/build.go:626
        	Error:      	Received unexpected error:
        	            	json: cannot unmarshal string into Go struct field mdT.buildx.build.status of type client.SolveStatus
        	Test:       	TestIntegration/TestBuildMetadataStatus/worker=docker-container/warnings
```

Seems to work fine locally with docker builder (BuildKit 0.13.2).